File: test-utils/src/main/java/ru/virgil/test_utils/fluent_request/RequestUtil.java
package ru.virgil.test_utils.fluent_request;

import com.fasterxml.jackson.databind.ObjectMapper;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@RequiredArgsConstructor
@Component
public class RequestUtil {

    private final Requester requester;
    private final ObjectMapper objectMapper;

    private BodyStepJson startJsonBuilding(RequestMethod requestMethod, String path) {
        RequestModel requestModel = new RequestModel();
        requestModel.setRequestMethod(requestMethod);
        requestModel.setUrl(path);
        return new BodyStepJson(requestModel, requester, objectMapper);
    }

    private BodyStepMultipartStart startMultipartBuilding(RequestMethod requestMethod, String path) {
        RequestModel requestModel = new RequestModel();
        requestModel.setRequestMethod(requestMethod);
        requestModel.setUrl(path);
        return new BodyStepMultipart(requestModel, requester, objectMapper);
    }

    public BodyStepJsonStart get(String path) {
        return startJsonBuilding(RequestMethod.GET, path);
    }

    public BodyStepJsonStart post(String path) {
        return startJsonBuilding(RequestMethod.POST, path);
    }

    public BodyStepJsonStart put(String path) {
        return startJsonBuilding(RequestMethod.PUT, path);
    }

    public BodyStepJsonStart delete(String path) {
        return startJsonBuilding(RequestMethod.DELETE, path);
    }

    public BodyStepMultipartStart postMultipart(String path) {
        return startMultipartBuilding(RequestMethod.POST_MULTIPART, path);
    }
}

File: tools/src/main/kotlin/ru/virgil/spring_tools/tools/testing/fluent_request/RequestUtil.kt
package ru.virgil.spring_tools.tools.testing.fluent_request

import com.fasterxml.jackson.databind.ObjectMapper
import org.springframework.stereotype.Component

@Deprecated("Use Fluent requests", ReplaceWith("Fluent"))
@Component
class RequestUtil(
    private val requester: Requester,
    private val objectMapper: ObjectMapper,
) {

    private fun startJsonBuilding(requestMethod: RequestMethod, path: String): BodyStepJson {
        val requestModel = RequestModel()
        requestModel.requestMethod = requestMethod
        requestModel.url = path
        return BodyStepJson(requestModel, requester, objectMapper)
    }

    private fun startMultipartBuilding(path: String): BodyStepMultipartStart {
        val requestModel = RequestModel()
        requestModel.requestMethod = RequestMethod.POST_MULTIPART
        requestModel.url = path
        return BodyStepMultipart(requestModel, requester, objectMapper)
    }

    fun get(path: String): BodyStepJsonStart = startJsonBuilding(RequestMethod.GET, path)

    fun post(path: String): BodyStepJsonStart = startJsonBuilding(RequestMethod.POST, path)

    fun put(path: String): BodyStepJsonStart = startJsonBuilding(RequestMethod.PUT, path)

    fun delete(path: String): BodyStepJsonStart = startJsonBuilding(RequestMethod.DELETE, path)

    fun postMultipart(path: String): BodyStepMultipartStart = startMultipartBuilding(path)

}
